terraform/modules/query-service-infra/sqs.tf

```diff
@@ -0,0 +1,54 @@
+
+resource "aws_sqs_queue" "query_queue" {
+  name                      = "dcp-query-data-input-queue-${var.deployment_stage}"
+//  Queue visibility timeout must be larger than (triggered lambda) function timeout
+  visibility_timeout_seconds = 900
+  message_retention_seconds = 86400
+  redrive_policy            = "{\"deadLetterTargetArn\":\"${aws_sqs_queue.deadletter_queue.arn}\",\"maxReceiveCount\":4}"
+
+}
+
+
+resource "aws_sqs_queue" "deadletter_queue" {
+  name                      = "dcp-query-data-input-deadletter-queue-${var.deployment_stage}"
+  message_retention_seconds = 1209600
+}
+
+resource "aws_sqs_queue_policy" "data_input_query_queue_access" {
+  queue_url = "${aws_sqs_queue.query_queue.id}"
+
+  policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Id": "sqspolicy",
+  "Statement": [
+  {
+      "Sid": "First",
+      "Effect": "Allow",
+      "Principal": "*",
+      "Action": "SQS:SendMessage",
+      "Resource": "arn:aws:sqs:*:*:*"
```

Comment:
I'd limit this just to the specific queue your lambda needs to access.

Reply:
👍